torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1929,6 +1932,16 @@ def hint():
                     source == symbol_to_source[expr][0]
                 ):
                     continue
+
+                # This logic excludes static values found on tensors from guarding, because
+                # dynamo's check_tensor_fn does that (see guards.cpp).
+                # However, for non tensor sources, we still need to guard here.
+                if ignore_static and isinstance(source, TensorPropertySource):
+                    maybe_static = self._maybe_evaluate_static(expr)
```

Comment:
Hmm, do you really need to maybe_evaluate_static here? Shouldn't we already have done all the simplification you need?

Reply:
You want the logic here to be symmetric with constraint violation logic which just checks concreteness directly